src/vs/editor/contrib/contextmenu/browser/contextmenu.ts

```diff
@@ -161,7 +161,18 @@ export class ContextMenuController implements IEditorContribution {
 
 		// get menu groups
 		const menu = this._menuService.createMenu(menuId, this._contextKeyService);
-		const groups = menu.getActions({ arg: model.uri });
+		const groups = menu.getActions({
+			arg: [model.uri, this._editor.getSelections()?.map((selection) => ({
+				start: {
+					line: selection.getStartPosition().lineNumber - 1,
+					character: selection.getStartPosition().column
+				},
+				end: {
+					line: selection.getEndPosition().lineNumber - 1,
+					character: selection.getEndPosition().column
+				}
+			}))]
+		});
```

Comment:
This is not OK because it is a conceptual layer breaker. It introduces extension host types to the renderer, e.g this new argument is wrong for all "internal" context menu actions and wrong for monaco-editor use-cases.